src/java.desktop/share/classes/sun/font/PhysicalFont.java

```diff
@@ -41,14 +42,9 @@ public abstract class PhysicalFont extends Font2D {
     protected Object nativeNames;
 
     public boolean equals(Object o) {
-        if (o == null || o.getClass() != this.getClass()) {
-            return false;
-        }
-        PhysicalFont other = (PhysicalFont)o;
-        return
-           (this.fullName.equals(other.fullName)) &&
-            ((this.platName == null && other.platName == null) ||
-             (this.platName != null && this.platName.equals(other.platName)));
+        return o instanceof PhysicalFont other
```

Comment:
The old implementation also checked that the runtime class of `other` is the same as that of `this`:
```suggestion
        return o instanceof PhysicalFont other
                && other.getClass() == this.getClass()
```

Reply:
Intentional change. There is only one class that extends PhysicalFont, which also happens to overrides PhysicalFont's equals. So checking via instanceof shouldn't mean any change. Also I think I forgot to add a `o == this ||` to some of them... Tomorrow's problem.